### PR TITLE
[RD2] Schedule Component recurring type bugfix & tests

### LINKED ISSUE #6169: Enhanced Recurring Donations: Validation error on the RD2_EntryForm when changing a Fixed Recurring Donation to an Open one where the "Number of Planned Installments" is greater than 1
### Summary of Issue

When a customer tries to change a Fixed Recurring Donation to an Open Recurring Donation by changing the Recurring Type field, they will receive a validation error and it will not be apparent how to fix the issue. The error they receive is:

"An error occurred while trying to update the record. Please try again.
Number of Planned Installments is for Fixed Recurring Donations only. Remove the value in Number of Planned Installments or change the Recurring Type to Fixed."

### Expected Result

The user should be able to change a Fixed Recurring Donation to Open without receiving the validation error or having to use one of the workarounds below. 

### Steps to Reproduce

**Scenario 1**

1. Create a Fixed-Length Recurring Donation
    - Donor Type: Contact
    - Date Established: Today
    - Contact: Test Contact
    - Amount: 5
    - Recurring Type: Fixed
    - Number of Planned Installments: 12
    - Recurring Period: Monthly
    - Day of Month: Today
    - Effective Date: Today
    - Recurring Donation Name: Test
2. Edit the existing record
    - Change Recurring Type to: Open
    - Click Save

**Scenario 2**

1. As per 1 above
2. Edit the existing record
     - Clear out the value of "Number of Planned Installments"
     - Change Recurring Type to: Open
     - Click Save

**Animation Illustrating the Issue**

RD Form UX issue

**Additional Notes**

- The "Number of Planned Installments" is a required field on the RD2_EntryForm when Fixed, and the field is then dynamically hidden when changing the Recurring Type to Open but does not keep the field value in Scenario 2.2 above
- The user will receive the validation error when trying to change the Recurring Type on any Fixed Recurring Donation that has a "Number of Planned Installments" is greater than 1 i.e if the "Number of Planned Installments" equals 1 a user will be able to successfully change the Recurring Type without receiving an error

### Workarounds

- Create a Quick Action on Recurring Donations to change from Fixed to Open (setting predefined values) and use this Quick Action instead of editing the record directly
- Within a record edit context, change "Number of Planned Installments" to 1, then change the Recurring Type to Open and save the record
- Change the "Number of Planned Installments" to 1 and save the record, then change the Recurring Type to Open and save the record

### PR DESCRIPTION
**Pull request by bdvorachek**
WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008F7nnIAC/view
# Critical Changes

# Changes
 - We made improvements to permissions security checks on the Recurring Donations entry form.

# Issues Closed
Closes #6169
# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: force-app/main/default/lwc/rd2EntryFormScheduleSection/rd2EntryFormScheduleSection.js [hunk 1]
```diff
@@ -557,6 +559,10 @@ export default class rd2EntryFormScheduleSection extends LightningElement {
                 }
             });
 
+        if(data[this.fields.recurringType.apiName] === RECURRING_TYPE_OPEN) {
+            data[this.fields.plannedInstallments.apiName] = null;
```
Comment (lparrott) [inline force-app/main/default/lwc/rd2EntryFormScheduleSection/rd2EntryFormScheduleSection.js:563]: This works great, allows the form to keep the Installment Number value if customers switch from Fixed to Open, and then decide to switch back.

File: force-app/main/default/lwc/rd2EntryFormScheduleSection/__tests__/rd2EntryFormScheduleSection.test.js [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+        beforeEach(async () => {
+            component = createElement('c-rd2-entry-form-schedule-section', { is: rd2EntryFormScheduleSection });
+            component.addEventListener('installmentschange', mockHandleInstallmentsChange);
+            controller = new Rd2EntryFormScheduleSectionTestController(component);
+            getRecurringSettings.mockResolvedValue({ ...mockGetRecurringSettings, defaultRecurringType: 'Fixed' });
+            document.body.appendChild(component);
+            await setupWires();
+        });
+
+        it('number of planned installments is not rendered', async () => {
```
Comment (lparrott) [inline force-app/main/default/lwc/rd2EntryFormScheduleSection/__tests__/rd2EntryFormScheduleSection.test.js:110]: Should this description be "is rendered"?

**Review by lparrott — APPROVED** [review]
Great work on this! Let's update the PR with details on the bug fix, and reference a Github issue if it exists.

File: force-app/main/default/lwc/rd2EntryFormScheduleSection/__tests__/rd2EntryFormScheduleSection.test.js (hunk 2 above)
Comment (npsp-reedestockton) [reply force-app/main/default/lwc/rd2EntryFormScheduleSection/__tests__/rd2EntryFormScheduleSection.test.js:110]: Same question as @lparrott.  Seems reversed...

**Review by npsp-reedestockton — CHANGES_REQUESTED** [review]
Really nicely done @bdvorachek.  Requesting a minor change on line 110 of rd2EntryFormScheduleSection.test.js - unless I am misreading the logic the test wording is reversed.

**Review by npsp-reedestockton — APPROVED** [review]

**Pull request merged**